### PR TITLE
Upload page improvements

### PR DESCRIPTION
**Pull request by sashasashasasha151**
## Related issue(s)

https://issues.serokell.io/issue/EDNA-48

#### Related changes (conditional)

- Tests
  - [ ] If I added new functionality, I added tests covering it.
  - [ ] If I fixed a bug, I added a regression test to prevent the bug from
        silently reappearing again.

- Documentation
  - [ ] I checked whether I should update the docs and did so if necessary:
    - [README](../tree/master/README.md)

#### Stylistic guide (mandatory)

- [ ] My commits comply with [the policy used in Serokell](https://www.notion.so/serokell/Commit-and-PR-policy-4cf98e1a910a415d86b5f2491d9af1af).
- [ ] My code complies with the [style guide](../tree/master/docs/code-style.md).

### CODE REVIEW

**Review by gromakovsky — APPROVED** [review]
I have tested it, seems to work. It doesn't fully resolve EDNA-48 though, because:
1. There is no way to undo "Reset" and no notification about successfully added file.
2. Maybe we'll have another button to reset only file.
3. There are still several differences with design, but it's ok to deal with them in another PR.

File: frontend/src/components/UploadArea/UploadArea.tsx [hunk 1]
```diff
@@ -50,10 +50,12 @@ const UploadArea: FunctionComponent<UploadingAreaProps> = ({ value, onChange }):
...
       >
-        <div className="uploadArea__innerRect">
-          <UploadSvg className="uploadArea__uploadSvg" />
-          {value ? value.name : <strong>Click to upload</strong>}
-        </div>
+        <UploadSvg className="uploadArea__uploadSvg" />
+        {value ? (
+          value.name
+        ) : (
+          <span className="uploadArea__description">Drop your file here, or click to upload</span>
```
Comment (pva701) [inline frontend/src/components/UploadArea/UploadArea.tsx:73]: No, it's not true. Dropping doesn't work. Stop following the design blindly, this is not the last resort. At least the functionality should correspond to the reality.

File: frontend/src/components/footer/Footer.tsx [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+import cn from "../../utils/bemUtils";
+import "./Footer.scss";
+
+const Footer: FunctionComponent = (): ReactElement => {
+  const footer = cn("footer");
+
+  return (
+    <div className={footer()}>
+      <div className={footer("container")}>
+        <span className={footer("description")}>Powered by Serokell</span>
```
Comment (gromakovsky) [inline frontend/src/components/footer/Footer.tsx]: I know the footer in the design says "Powered by Serokell", but people say it makes little sense:
* https://serokell.slack.com/archives/C01EASAGPKN/p1612979249086100
* https://serokell.slack.com/archives/CCK72LXUH/p1618581789010500

Looking at YouTrack as an example
image
I propose to write "Edna—the data analysis tool for researchers by Serokell". I'll ask to update the design accordingly.

**Comment by pva701** [comment]
You forgot to disable outline for textarea.

**Review by pva701 — APPROVED** [review]

**Comment by gromakovsky** [comment]
One more minor thing: apparently according to design "Reset" button should be disabled initially (when all inputs are empty). If it's easy to do, please do it.

Also tomorrow there will be additional functionality: reset only the file, not all fields. But let's do it separately and merge this PR without it.

File: frontend/src/components/UploadArea/UploadArea.tsx [hunk 3]
```diff
@@ -12,6 +12,20 @@ const UploadArea: FunctionComponent<UploadingAreaProps> = ({ value, onChange }):
   const uploadInputRef = useRef<HTMLInputElement | null>(null);
   const setParsedFile = useSetRecoilState(excelFileAtom);
 
+  const handleFile = async (file: File) => {
+    onChange(file);
+    setParsedFile({ state: "uploading", progress: 0 });
+    try {
+      const targets = await Api.parseExcelFile(file, () => {});
```
Comment (pva701) [inline frontend/src/components/UploadArea/UploadArea.tsx]: oh, it's my bad. this callback shouldn't be empty. there should be updating `p => setParsedFile({ state: "uploading", progress: p });`

File: frontend/src/components/UploadArea/UploadArea.tsx [hunk 4]
```diff
@@ -49,6 +52,17 @@ const UploadArea: FunctionComponent<UploadingAreaProps> = ({ value, onChange }):
             uploadInputRef.current?.click();
           }
         }}
+        onDragOver={async e => {
+          e.preventDefault();
+        }}
+        onDrop={async e => {
+          e.preventDefault();
+          e.stopPropagation();
+          if (e.dataTransfer.files && e.dataTransfer.files.length > 0) {
```
Comment (gromakovsky) [inline frontend/src/components/UploadArea/UploadArea.tsx]: Does it permit uploading more than 1 file? I have no idea what I am writing, but shouldn't we check that `length == 1`?

**Pull request merged**